test/testtools/testtools.go

```diff
@@ -0,0 +1,69 @@
+package testtools
+
+import (
+	"github.com/fossas/fossa-cli/exec"
+
+	git "gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+)
+
+type Initializer interface {
+	// FixtureDirectory returns the directory of the test project fixtures
+	FixtureDirectory() string
+	// BuildAll fetches source code and runs all install/build scrips necessary for analysis
+	BuildAll() error
+	// RemoveAll removes all artifacts from BuildAll
+	RemoveAll() error
+}
+
+// SetupTests executes the initializer's build all and returns a function to be deferred for teardown. SetupTests and it's returned function are meant to be called once per analyzer type
+func SetupTests(initializer Initializer) (func(), error) {
+	err := initializer.BuildAll()
```

Comment:
It would make more sense for the consumer to call `initializer.BuildAll()` and then `defer initializer.RemoveAll()`.